src/docfx/build/Build.cs

```diff
@@ -36,33 +37,23 @@ private static async Task BuildDocset(string docsetPath, string outputPath, Comm
                 try
                 {
                     // load and trace entry repository
-                    var repositoryProvider = new RepositoryProvider(docsetPath, options);
+                    var repositoryProvider = new RepositoryProvider(docsetPath, options, () => restoreGitMap, () => config);
```

Comment:
config can be always loaded from current repository if we don;t need to consider the case that loc repo dones't have the config.

Reply:
https://github.com/dotnet/docfx/blob/b135d604cc5bc0f778cba42e3f0c9074cd6bde3b/src/docfx/config/ConfigLoader.cs#L111
`ConfigLoader` still depends on `RepositoryProvider` though